src/cpu/minor/ticked.hh

```diff
@@ -39,23 +39,20 @@
  * @file
  *
  *  Base classes for ClockedObjects which have evaluate functions to
- *  look like clock ticking operations.  TickedObject attaches gem5's event
- *  queue to Ticked to apply actual scheduling.
+ *  look like clock ticking operations.
  */
 
-#ifndef __SIM_TICKED_OBJECT_HH__
-#define __SIM_TICKED_OBJECT_HH__
+#ifndef __SIM_TICKED_HH__
+#define __SIM_TICKED_HH__
 
 #include "sim/clocked_object.hh"
 
-class TickedObjectParams;
-
 /** Ticked attaches gem5's event queue/scheduler to evaluate
  *  calls and provides a start/stop interface to ticking.
  *
  *  Ticked is not a ClockedObject but can be attached to one by
  *  inheritance and by calling regStats, serialize/unserialize */
-class Ticked : public Serializable
+class Ticked : public Serializable, public Stats::Group
```

Comment:
Does this need to be a Stats::Group? Would it make more sense to pass the CPU in as the parent to the struct?

Reply:
Yeah, it does not need to inherit from Stats::Group. I have pushed a change to do that. Ticked's implementation already seems tied to minor CPU, so I don't think this will be use in other ways